### PR TITLE
Remove cronjob that checks local versions every 10 minutes - only check when required

### PR DESCRIPTION
**Pull request by jpgpi250**
Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

## Thank you for your contribution to the Pi-hole Community! 

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**
 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions

---
- **What does this PR aim to accomplish?:**

see https://discourse.pi-hole.net/t/pi-hole-stoppen-zwecks-backup/57989/10

- eliminate updatechecker local from the pihole cronjob
- add updatecheck to piholeDebug.sh
- check if api.github.com returns a valid IP address

- **How does this PR accomplish the above?:**
changed:
- pihole.cron
- piholeCheckout.sh
- piholeDebug.sh
- updatecheck.sh

- **What documentation changes (if any) are needed to support this PR?:**

none

**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: advanced/Scripts/updatecheck.sh [hunk 1]
```diff
@@ -44,19 +44,32 @@ if [[ "$2" == "remote" ]]; then
         sleep 30
     fi
 
-    GITHUB_CORE_VERSION="$(curl -s 'https://api.github.com/repos/pi-hole/pi-hole/releases/latest' 2> /dev/null | jq --raw-output .tag_name)"
+    # Source the COL_TABLE
+    # shellcheck disable=SC1091
+    . /opt/pihole/COL_TABLE
+
+    apidomain="api.github.com"
```
Comment (PromoFaux) [inline advanced/Scripts/updatecheck.sh:51]: Can I request that we remove this specific changeset for the time being - and address in a separate PR if necessary? 

Currently I'm of the opinion that checking if `api.github.com` is resolvable is an unnecessary addition to code complexity with very little gain

**Review by PromoFaux — CHANGES_REQUESTED** [review]

File: advanced/Scripts/updatecheck.sh (hunk 1 above)
Comment (jpgpi250) [reply advanced/Scripts/updatecheck.sh]: removed...

File: advanced/Scripts/piholeDebug.sh [hunk 2]
```diff
@@ -41,6 +41,9 @@ else
     #OVER="\r\033[K"
 fi
 
+# Force an update of the updatechecker
```
Comment (PromoFaux) [inline advanced/Scripts/piholeDebug.sh:44]: This check is not necassary, as we only source `/etc/pihole/versions` to test for the value of `DOCKER_VERSION` (if it exists) in order to show that the debug run was done on a docker install.

The rest of the component versions are worked out in `compare_local_version_to_git_version`.

File: advanced/Scripts/piholeCheckout.sh [hunk 3]
```diff
@@ -42,6 +42,11 @@ warning1() {
     esac
 }
 
+updateCheckFunc() {
+/opt/pihole/updatecheck.sh
```
Comment (PromoFaux) [inline advanced/Scripts/piholeCheckout.sh:46]: Indent the code inside the function please :)

File: advanced/Scripts/piholeCheckout.sh [hunk 4]
```diff
@@ -137,6 +142,8 @@ checkout() {
             exit 1
         fi
         checkout_pull_branch "${PI_HOLE_FILES_DIR}" "${2}"
+        # Force an update of the updatechecker
+        updateCheckFunc
```
Comment (PromoFaux) [inline advanced/Scripts/piholeCheckout.sh:146]: Not needed as it is already run during the install script (which is run when core branches are checked out)

**Review by PromoFaux — CHANGES_REQUESTED** [review]

File: advanced/Scripts/piholeDebug.sh (hunk 2 above)
Comment (jpgpi250) [reply advanced/Scripts/piholeDebug.sh]: removed

File: advanced/Scripts/piholeCheckout.sh (hunk 3 above)
Comment (jpgpi250) [reply advanced/Scripts/piholeCheckout.sh]: done

File: advanced/Scripts/piholeCheckout.sh (hunk 4 above)
Comment (jpgpi250) [reply advanced/Scripts/piholeCheckout.sh]: removed

**Comment by PromoFaux** [comment]
One last request, can you squash all these commits back down to one, please? If you need help doing that, let me know!

**Comment by jpgpi250** [comment]
> One last request, can you squash all these commits back down to one, please? If you need help doing that, let me know!

using github desktop here, squash and merge is not available (greyed out)

image

**Comment by PromoFaux** [comment]
I'll do some magic. One moment please

**Comment by jpgpi250** [comment]
OK now?

**Comment by PromoFaux** [comment]
Ah! No, you've pushed back the commits that I squashed. I'll re-do what I did before (which was basically rebase on development and then squash and force push to your fork) - and after that you don't need to do anything

**Comment by PromoFaux** [comment]
What you might need to do on your end is a pull to your fork's development - but I'm not familiar with Github desktop enough to say either way. I use VSCode with Github extensions

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**